lib/ansible/modules/windows/win_disk_facts.ps1

```diff
@@ -0,0 +1,181 @@
+                        deduplication_enabled = $vdisk.IsDeduplicationEnabled
+                        enclosure_aware = $vdisk.IsEnclosureAware
+                        manual_attach = $vdisk.IsManualAttach
+                        snapshot = $vdisk.IsSnapshot
+                        tiered = $vdisk.IsTiered
+                        physical_sector_size = "$($vdisk.PhysicalSectorSize / 1KB)kb"
+                        logical_sector_size = "$($vdisk.LogicalSectorSize)byte"
+                        available_copies = "$([string]$vdisk.NumberOfAvailableCopies)"
+                        columns = $vdisk.NumberOfColumns
+                        groups = $vdisk.NumberOfGroups
+                        physical_disk_redundancy = $vdisk.PhysicalDiskRedundancy
+                        read_cache_size = $vdisk.ReadCacheSize
+                        request_no_spof = $vdisk.RequestNoSinglePointOfFailure
+                        resiliency_setting_name = $vdisk.ResiliencySettingName
+                        object_id = $vdisk.ObjectId
+                        unique_id_format = $vdisk.UniqueIdFormat
+                        unique_id = $vdisk.UniqueId
+                    }
+                }
+            }
+            $disk_results.number = $disk.Number
+            $disk_results.size = "$($disk.Size / 1GB)gb"
+            $disk_results.bus_type = $disk.BusType
+            $disk_results.friendly_name = $disk.FriendlyName
+            $disk_results.partition_style = $disk.PartitionStyle
+            $disk_results.partition_count = $disk.NumberOfPartitions
+            $disk_results.operational_status = $disk.OperationalStatus
+            $disk_results.sector_size = "$($disk.PhysicalSectorSize)byte"
+            $disk_results.read_only = $disk.IsReadOnly
+            $disk_results.bootable = $disk.IsBoot
+            $disk_results.system_disk = $disk.IsSystem
+            $disk_results.clustered = $disk.IsClustered
+            $disk_results.manufacturer = $disk.Manufacturer
+            $disk_results.model = $disk.Model
+            $disk_results.firmware_version = $disk.FirmwareVersion
+            $disk_results.location = $disk.Location
+            $disk_results.serial_number = $disk.SerialNumber
+            $disk_results.unique_id = $disk.UniqueId
+            $disk_results.guid = "$([string]$disk.Guid)"
+            $disk_results.path = $disk.Path
+            $parts = Get-Partition -DiskNumber $($disk.Number) -ErrorAction SilentlyContinue
+            if ($parts) {
+                $disk_results["partitions"]  += @()
+                foreach ($part in $parts) {
+                    $partition_results  = @{
+                        number = $part.PartitionNumber
+                        size = "$($pSize = "{0:N3}" -f ($part.Size /1GB))$($pSize)gb"
+                        type = $part.Type
+                        drive_letter = "$([string]$part.DriveLetter)"
```

Comment:
When this is null, the drive letter is coming up as `\u0000` (`\x00`), if you remove the casting to a string this return `null` in the JSON result which is better.

Reply:
Ok, thought it is better to have an empty string than null.
I have changed this now.